streams/src/test/java/org/apache/kafka/streams/state/internals/CachingPersistentWindowStoreTest.java

```diff
@@ -1039,58 +1041,42 @@ public void shouldNotThrowInvalidBackwardRangeExceptionWithNegativeFromKey() {
     @Test
     public void shouldCloseCacheAndWrappedStoreAfterErrorDuringCacheFlush() {
         setUpCloseTests();
-        EasyMock.reset(cache);
-        cache.flush(CACHE_NAMESPACE);
-        EasyMock.expectLastCall().andThrow(new RuntimeException("Simulating an error on flush"));
-        cache.close(CACHE_NAMESPACE);
-        EasyMock.replay(cache);
-        EasyMock.reset(underlyingStore);
-        underlyingStore.close();
-        EasyMock.replay(underlyingStore);
+        doThrow(new RuntimeException("Simulating an error on flush")).doNothing().when(cache).flush(CACHE_NAMESPACE);
 
         assertThrows(RuntimeException.class, cachingStore::close);
-        EasyMock.verify(cache, underlyingStore);
+        verify(cache).flush(CACHE_NAMESPACE);
+        verify(cache).close(CACHE_NAMESPACE);
+        verify(underlyingStore).close();
```

Comment:
Do you think there are any additional benefits if we use Mockito's `InOrder` interface to verify these method calls?  (same question applies to other changed `verify` calls in this class)

<img width="597" alt="image" src="https://user-images.githubusercontent.com/9079844/184432082-93586a57-0bef-46a2-b39b-1d58062123e3.png">

Link to docs:
- https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/InOrder.html

Reply:
I am ambivalent. My reasoning would be to use InOrder to confirm that records are consumed in order, much like what Bruno pointed out in https://github.com/apache/kafka/pull/12492#discussion_r944369076 rather than in this case. If you would like stronger assertions I can add them. What is your preference?

Reply:
I think it makes sense to use `InOrder` here since we want to flush a cache/store before closing it.

Reply:
Hopefully I have addressed this in subsequent commits.